src/modules/avformat/producer_avformat.c

```diff
@@ -1934,20 +1930,22 @@ static int producer_get_image( mlt_frame frame, uint8_t **buffer, mlt_image_form
 						else
 						{
 #if USE_HWACCEL
-							if ( self->hw_device_ctx && self->video_frame->format == self->hw_pix_fmt )
+							if (self->hwaccel.device_ctx && self->video_frame->format == self->hwaccel.pix_fmt)
 							{
-								int transfer_data_result = av_hwframe_transfer_data( self->sw_video_frame, self->video_frame, 0 );
-								if( transfer_data_result < 0 ) 
+								AVFrame *sw_video_frame = av_frame_alloc();
+								int transfer_data_result = av_hwframe_transfer_data(sw_video_frame, self->video_frame, 0);
```

Comment:
`sw_video_frame` doesn't have any use outside of this code section, and I don't imagine this would change. Removing it from producer context seems like a good idea.

Reply:
OK, but like above, I think we should instead use `av_frame_free()` instead of`av_frame_unref()`.